src/derivative_wrappers.jl

```diff
@@ -868,6 +911,29 @@ function accumulate_cost!(dλ, y, p, t, S::TS,
     return nothing
 end
 
+function accumulate_cost(dλ, y, p, t, S::TS,
+                         dgrad = nothing) where {TS <: SensitivityFunction}
+    @unpack dg, dg_val, g, g_grad_config = S.diffcache
+    if dg !== nothing
+        if !(dg isa Tuple)
+            dg(dg_val, y, p, t)
```

Comment:
do you have a test case where this is actually triggered (i.e, a continuous and a discrete cost contribution)? I'd think it makes more sense to write this part also fully non-mutating.